packages/inference/src/InferenceClient.ts

```diff
@@ -40,34 +33,25 @@ export class InferenceClient {
 	}
 
 	/**
-	 * Returns copy of InferenceClient tied to a specified endpoint.
+	 * Returns instance of InferenceClient tied to a specified endpoint.
+	 *
+	 * For backward compatibility mostly.
 	 */
-	public endpoint(endpointUrl: string): InferenceClientEndpoint {
-		return new InferenceClientEndpoint(endpointUrl, this.accessToken, this.defaultOptions);
-	}
-}
-
-export class InferenceClientEndpoint {
-	constructor(endpointUrl: string, accessToken = "", defaultOptions: Options = {}) {
-		accessToken;
-		defaultOptions;
-
-		for (const [name, fn] of Object.entries(tasks)) {
-			Object.defineProperty(this, name, {
-				enumerable: false,
-				value: (params: RequestArgs, options: Options) =>
-					// eslint-disable-next-line @typescript-eslint/no-explicit-any
-					fn({ ...params, accessToken, endpointUrl } as any, { ...defaultOptions, ...options }),
-			});
-		}
+	public endpoint(endpointUrl: string): this {
+		this.defaultOptions.endpointUrl = endpointUrl;
```

Comment:
I don't think it should override this, it should return a new instance. 

Eg if someone has:

```ts
const hf = new InferneceClient(token);

const endpoint1 = hf.endpoint("...");
const endpoint2 = hf.endpoint("...");
```

it will break stuff.

Honestly this PR should just add the possibility to add `endpointsUrl` in the constructor of `InferenceClient`, and pass it in the `args` in that case, that 's all.

Reply:
but ok for returning a new instance, i'll make the change

Reply:
f86dbea25fb442c68e5729a70c4087cf9e8a51e7